dbt/include/global_project/macros/adapters/timestamps.sql

```diff
@@ -15,6 +15,14 @@
     {{ current_timestamp() }}
 {% endmacro %}
 
+{% macro snapshot_get_time_data_type() %}
```

Comment:
nit: could we rename this to have a `get_` prefix similar to `get_updated_at_column_data_type`? I realize it's a bit wordy especially with `get` already being in the name but I think it would help to differentiate this method from the variable its assigned to later on when we do: 
```
{% set get_time_data_type = snapshot_get_time_data_type() %}
```
could instead be: 
```
{% set snapshot_get_time_data_type = get_snapshot_get_time_data_type() %}
```